src/Stack/Types/Package.hs

```diff
@@ -92,29 +92,104 @@ newtype ExeName = ExeName { unExeName :: Text }
+--    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:buildTypeRaw buildTypeRaw function (and buildType, but it's about the same)>
 data Package =
-  Package {packageName :: !PackageName                    -- ^ Name of the package.
-          ,packageVersion :: !Version                     -- ^ Version of the package
-          ,packageLicense :: !(Either SPDX.License License) -- ^ The license the package was released under.
-          ,packageFiles :: !GetPackageFiles               -- ^ Get all files of the package.
-          ,packageDeps :: !(Map PackageName DepValue)     -- ^ Packages that the package depends on, both as libraries and build tools.
-          ,packageUnknownTools :: !(Set ExeName)          -- ^ Build tools specified in the legacy manner (build-tools:) that failed the hard-coded lookup.
-          ,packageAllDeps :: !(Set PackageName)           -- ^ Original dependencies (not sieved).
-          ,packageGhcOptions :: ![Text]                   -- ^ Ghc options used on package.
-          ,packageCabalConfigOpts :: ![Text]              -- ^ Additional options passed to ./Setup.hs configure
-          ,packageFlags :: !(Map FlagName Bool)           -- ^ Flags used on package.
-          ,packageDefaultFlags :: !(Map FlagName Bool)    -- ^ Defaults for unspecified flags.
-          ,packageLibraries :: !PackageLibraries          -- ^ does the package have a buildable library stanza?
-          ,packageInternalLibraries :: !(Set Text)        -- ^ names of internal libraries
-          ,packageTests :: !(Map Text TestSuiteInterface) -- ^ names and interfaces of test suites
-          ,packageBenchmarks :: !(Set Text)               -- ^ names of benchmarks
-          ,packageExes :: !(Set Text)                     -- ^ names of executables
-          ,packageOpts :: !GetPackageOpts                 -- ^ Args to pass to GHC.
-          ,packageHasExposedModules :: !Bool              -- ^ Does the package have exposed modules?
-          ,packageBuildType :: !BuildType                 -- ^ Package build-type.
+  Package {packageName :: !PackageName
+          -- ^ Name of the package.
+          -- This is PackageIdentifier pkgName in cabal. 
+          ,packageVersion :: !Version
+          -- ^ Version of the package.
+          -- This is (maybe) PackageIdentifier pkgVersion in cabal.
+          ,packageLicense :: !(Either SPDX.License License)
+          -- ^ The license the package was released under.
+          -- This is PackageDescription licenseRaw in cabal. 
+          ,packageFiles :: !GetPackageFiles
+          -- ^ Get all files of the package.
+          -- Absent in cabal.
+          ,packageDeps :: !(Map PackageName DepValue)
+          -- ^ Packages that the package depends on, both as libraries and build tools.
+          -- Likely derived from cabal:
```

Comment:
It looks a bit odd to have "likely" in haddocks for our own definitions...

Reply:
Yes, but it's better than nothing. I didn't actually track all the fields down, because there are a number of hacks on pretty much all of them, the idea of this is to understand why everything is needed and how we can better address these needs. I agree it's not the best, maybe we should wait for the actual component changes on Packge to merge this then.